graphql/src/store/prefetch.rs

```diff
@@ -543,17 +543,17 @@ fn object_or_interface_by_name<'a>(
     }
 }
 
-fn execute_selection_set(
+fn execute_selection_set<'a>(
     resolver: &StoreResolver<impl Store>,
-    ctx: &ExecutionContext<impl Resolver>,
+    ctx: &'a ExecutionContext<impl Resolver>,
     mut parents: Vec<Node>,
-    selection_set: &q::SelectionSet,
+    selection_sets: Vec<&'a q::SelectionSet>,
```

Comment:
Just out of curiosity: why a `Vec` here instead of an iterator?

Reply:
There's a story here, I got a `recursion limit reached` error from rustc when I tried using a generic here, which indicates it somehow lead to generating an infinitely recursive type or something. I could not understand why that error happened so I switched to vec which fixed it.